File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
         
        double sum = accumulate(nums.begin() , nums.begin()+k  , 0);
        
        double avg = sum/k ;
        
         for(int i = k ; i < nums.size() ; ++i){
             sum += nums[i] - nums[i-k];
             avg = max(avg , sum/k);
             
         }
        
        return avg;
    }
};

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        
        n = len(nums)
        maxavg = -inf
        left = 0
        Sum = 0
        for i in range(n):
            Sum += nums[i]
            
            if i + 1 >= k:
                maxavg = max(maxavg , Sum/k)
                Sum -= nums[left]
                left +=1
                
        return maxavg
                